packages/packages/commonwell-sdk/src/client/document.ts

```diff
@@ -28,11 +29,21 @@ export async function query(
   } catch (err) {
     throw new CommonwellError(`Error querying documents`, err, additionalInfo);
   }
+  return response;
+}
+
+export async function query(
```

Comment:
Only if we keep both query and queryFull: those are almost the same function, only diff is the schema we're using, right? We could move that logic to a generic function that gets the schema so we have the code in one place, and `query` and `queryFull` call that function sending the schema to be used to parse the response.

Reply:
No worries, I thought it would make it simpler. NABD though.

Reply:
AFAIK it seems like to infer the type properly we'd have to start getting into type predicates (which seems like a bit much overhead for this): https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates